aws/resource_aws_sagemaker_code_repository.go

```diff
@@ -0,0 +1,205 @@
+	return &schema.Resource{
+		Create: resourceAwsSagemakerCodeRepositoryCreate,
+		Read:   resourceAwsSagemakerCodeRepositoryRead,
+		Update: resourceAwsSagemakerCodeRepositoryUpdate,
+		Delete: resourceAwsSagemakerCodeRepositoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"code_repository_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 63),
+			},
+			"git_config": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"repository_url": {
+							Type:     schema.TypeString,
+							ForceNew: true,
+							Required: true,
+						},
+						"branch": {
+							Type:         schema.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1024),
+						},
+						"secret_arn": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSagemakerCodeRepositoryCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	name := d.Get("code_repository_name").(string)
+
+	createOpts := &sagemaker.CreateCodeRepositoryInput{
+		CodeRepositoryName: aws.String(name),
+		GitConfig:          expandSagemakerCodeRepositoryGitConfig(d.Get("git_config").([]interface{})),
+	}
+
+	log.Printf("[DEBUG] sagemaker code repository create config: %#v", *createOpts)
+	_, err := conn.CreateCodeRepository(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Sagemaker code repository: %w", err)
+	}
+
+	d.SetId(name)
+
+	return resourceAwsSagemakerCodeRepositoryRead(d, meta)
+}
+
+func resourceAwsSagemakerCodeRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	describeNotebookInput := &sagemaker.DescribeCodeRepositoryInput{
+		CodeRepositoryName: aws.String(d.Id()),
+	}
+	CodeRepository, err := conn.DescribeCodeRepository(describeNotebookInput)
+	if err != nil {
+		if isAWSErr(err, "ValidationException", "Cannot find CodeRepository") {
+			d.SetId("")
+			log.Printf("[WARN] Unable to find SageMaker code repository (%s); removing from state", d.Id())
+			return nil
+		}
+		return fmt.Errorf("error finding sagemaker code repository (%s): %w", d.Id(), err)
+
+	}
+
+	if err := d.Set("code_repository_name", CodeRepository.CodeRepositoryName); err != nil {
+		return fmt.Errorf("error setting code_repository_name for sagemaker code repository (%s): %w", d.Id(), err)
+	}
+
+	if err := d.Set("arn", CodeRepository.CodeRepositoryArn); err != nil {
+		return fmt.Errorf("error setting arn for sagemaker code repository (%s): %w", d.Id(), err)
+	}
```

Comment:
No need to check `err` for simple attribute values.

Reply:
done